src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -237,4 +243,40 @@ private CompletableFutures() {
         .thenApply(ignored ->
                        function.apply(af.join(), bf.join(), cf.join(), df.join(), ef.join()));
   }
+
+  /**
+   * Periodically poll an external resource until it returns a non-empty result.
+   *
+   * The polling task should return Optional.empty() until it becomes available, and then
+   * Optional.of(result).
+   *
+   * If the polling task throws an exception or returns null, that will cause the result future to
+   * complete exceptionally.
+   *
+   * @param pollingTask the polling task.
+   * @param frequency the frequency to run the polling task at.
+   * @param executorService the executor service to schedule the polling task on.
+   * @return a future completing to the result of the polling task once that becomes available.
+   */
+  public static <T> CompletableFuture<T> poll(
+      final Supplier<Optional<T>> pollingTask,
+      final Duration frequency,
+      final ScheduledExecutorService executorService) {
+    final CompletableFuture<T> result = new CompletableFuture<>();
+    final ScheduledFuture<?> scheduled = executorService.scheduleAtFixedRate(
+        () -> pollTask(pollingTask, result), 0, frequency.toMillis(), TimeUnit.MILLISECONDS);
+    result.whenComplete((r, ex) -> scheduled.cancel(true));
```

Comment:
@dflemstr I don't think this is a huge issue. They way I read the docs/code it looks like in the current state (`removeOnCancelPolicy==false`) the task will stay in the queue for at most the polling frequency time and will be collected at that point. I tried it like this:

```java
  @Test
  public void poll_notRunningAfterCancel() throws Exception {
    final ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
    final CompletableFuture<String> future = poll(Optional::empty, Duration.ofMillis(2), executor);

    future.cancel(true);
    Thread.sleep(10);

    final List<Runnable> running = executor.shutdownNow();
    assertThat(running, is(empty()));
  }
```
This will work if you block the testcase for a few ms (using the `Thread.sleep`), which gives the `ScheduledExecutorService` time to collect the task within the poll duration. Removing the sleep does indeed break the test.